### DISCUSSION TITLE
Cannot access the container running in VM of VirtualBox

### DISCUSSION THREAD

**Discussion by dunmorogher** [containers/podman#23399] [General]
Hello! 
I am new to Podman and I had Podman installed on the VM (Alma Linux 9.4) with VirtualBox 7.0.18. I configured two network interfaces - One is of NAT Network; another is Host-Only (which I am using for communicating between Host machine and VM)

After all setups, I created one container of `httpd` service with root user (please see it below...)

![image](https://github.com/user-attachments/assets/54c10684-b860-4589-a904-06b629083eda)

I am able to access it from the UI inside of VM with using both `localhost:8080` or `http://192.168.123.210:8080`

![image](https://github.com/user-attachments/assets/96981fa5-3aa3-4b68-a753-9a5ec923505d)

![image](https://github.com/user-attachments/assets/f751b9f0-ef65-4506-aee4-f70a9b1c916b)

However, I could not access it from the Host machine

![image](https://github.com/user-attachments/assets/9c082b28-aad1-45c6-9023-a97216a4f96d)

I am sure the connection between Host machine and VM is working, as I can access the Cockpit in the VM from Host machine

![image](https://github.com/user-attachments/assets/78f66cec-c98d-469f-8e51-b8fcb217e655)

I do not know where I probably made wrong...if you could help and suggest on my issue?

Thanks in advance!

**Comment by rhatdan** [comment]
Did you configure the host firewall to open those ports.  This is something gvproxy is handling in podman machine.

**Reply by dunmorogher** [reply]
Hello. I found out it why...I installed both Docker and Podman to that VM. After uninstalled Docker, it worked correctly...it seems Podman and Docker are not able to exist together. :-(

**Reply by rhatdan** [reply]
They should be able to, unless they are fighting over iptables rules.

**Reply by dunmorogher** [reply]
Yeah, but I did not check it with iptable rules; just uninstalled Docker then the issue was gone...

**Reply by rhatdan** [reply]
Docker manipulates the iptables rules itself.  Podman does not, because we feel this is security feature that an admin should be in charge of and not as easy to open up vulnerabilties.

**Reply by Luap99** [reply]
We also configure the firewall by default to allow port forwarding as root. As rootless we never do this because we cannot as we have no privileges to do so.

The problem with docker is that they actually setup a deny all rule, https://docs.docker.com/network/packet-filtering-firewalls/#docker-on-a-router which then might block other traffic. It has been a while since I looked at the docker rules but most likely that our rules just conflict in certain unexpected ways because of that.

**Reply by dunmorogher** [reply]
@rhatdan, @Luap99 - I see. Thank you for explaining!
